modules/gdscript/gdscript.cpp

```diff
@@ -1102,6 +1102,9 @@ GDScript *GDScript::find_class(const String &p_qualified_name) {
 		// Script path could have a class path separator("::") in it.
 		class_names = p_qualified_name.trim_prefix(get_root_script()->path).split("::");
 		result = get_root_script();
+	} else if (path.begins_with(get_root_script()->path)) {
+		class_names = path.trim_prefix(get_root_script()->path).split("::");
+		result = get_root_script();
```

Comment:
I'm not sure, but the condition looks as always true to me. `p_qualified_name` is not used at all in the branch, this is weird.

Reply:
Thanks for looking at this! I tried to explain what I found while debugging [here](https://github.com/godotengine/godot/issues/83542#issuecomment-1782628013).

If you can provide more details, I can dive back in.

"_The condition looks as always true_" - Which condition?
_"`p_qualified_name` is not used_" - What exactly do you mean by "not used" and what is weird about it?

Reply:
> Which condition?

`path.begins_with(get_root_script()->path)`. `path` is the script path, for inner classes it may additionally contain suffixes. `get_root_script()` returns the topmost script that owns the inner class scripts.

> What exactly do you mean by "not used" and what is weird about it?

This method checks `p_qualified_name`, every other branch has `p_qualified_name`, but not this one, that's weird. I haven't tested this, but it might break the `_owner != nullptr` branch.

Reply:
>  I haven't tested this, but it might break the _owner != nullptr branch.

How can I test for this? 
Adding some inner classes to my MRP?
I run `--test --test-suite="*GDScript*"` and everything seems to be fine.
![image](https://github.com/godotengine/godot/assets/41547570/57d3ef55-6eca-4100-8c68-234ba695c1f2)

Reply:
Successful tests do not guarantee the absence of regressions, since the tests do not cover all possible scenarios. I added this PR for discussion at the GDScript team meeting or maybe @vnen will review it offline.

Reply:
It is odd to me as well. This is a function to find a class identified by `p_qualified_name`, so the branch not checking the argument cannot reliably find the requested class. As @dalexeev said, this will always be true (the path of an inner class will always start with the path of its root script), so the other conditions will never be checked.

Given that, I don't think this is a proper solution.